website/client/js/services/userServices.js

```diff
@@ -625,7 +633,7 @@ angular.module('habitrpg')
           var isStaticOrSocial = $window.location.pathname.match(/^\/(static|social)/);
           if (!isStaticOrSocial){
             localStorage.clear();
-            $location.path('/logout');
+            window.location.href = '/logout';
```

Comment:
This is why the build is failing. Karma doesn't like full page redirects. you could use $window instead.